src/cmd/top/joblist_pane.c

```diff
@@ -370,15 +370,33 @@ void joblist_pane_set_current (struct joblist_pane *joblist, bool next)
     if (joblist->current != FLUX_JOBID_ANY)
         index = lookup_jobid_index (joblist->jobs, joblist->current);
 
+    /* find next valid index */
     njobs = json_array_size (joblist->jobs);
-    if (next && index == njobs - 1)
-        index = -1;
-    else if (!next && index <= 0)
-        index = njobs;
+    while (1) {
```

Comment:
Just an idle question. Would it be simpler to create a separate "filtered" list of jobs and use the simpler selection code to select among the displayed jobs? It might be useful going forward if `flux top` ever filters jobs with something other than queues (e.g. by user), since the same selection code will work no matter how the list is filtered.

Reply:
at the time I just didn't want to make a copy of the whole joblist, b/c currently its an incref copy from what was returned from the joblist query.

```
    if (flux_rpc_get_unpack (f, "{s:o}", "jobs", &jobs) < 0) {                                                                              
        if (errno != ENOSYS)                                                                                                                
            fatal (errno, "error decoding job-list.list RPC response");                                                                     
        flux_future_destroy (f);                                                                                                            
        return;                                                                                                                             
    }                                                                                                                                       
    json_decref (joblist->jobs);                                                                                                            
    joblist->jobs = json_incref (jobs);         
```

But you make a good point that such a filtered list is a good "investment" for easier future filtering.  Lemme take a crack at it.

Reply:
Could you use a zlistx as a filtered "view" into the list of all jobs? Not at a computer, sorry if that's a silly suggestion

Reply:
Looking through the code, I'd like to look at the "filtered job list" as a refactor after PR #5052.  With the queue abstraction that I placed in that PR I think it'd be easier.  I did ponder refactoring both this PR and that PR to put the queue abstraction "earlier" in the commit series, but I figure it's probably less work overall to do it after the two PRs together.

Reply:
Seems reasonable to me!